src/model.rs

```diff
@@ -84,10 +88,20 @@ pub enum ScuttleButtMessage {
     Ack { delta: Delta },
 }
 
-#[derive(Default, Serialize)]
 pub struct NodeState {
     max_version: u64,
     key_values: HashMap<String, VersionedValue>,
+    last_heartbeat: Instant,
```

Comment:
If I understand phi accrual failure detection correctly, all one would have to do to implement it with my current approach is store an array of Instants here. Then it should be trivial to compute expected heartbeat arrival time in the `living_nodes` function.

Reply:
I believe this is correct.